### PR TITLE
feat(examples): Add core-utils

### PR DESCRIPTION
**Pull request by feohuman**
Add basic linux utils (ls, cat, pwd, uniq) in unikernel format, running on debian:bookworm.

### CODE REVIEW

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

File: examples/core-utils/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM debian:bookworm AS build 
+
+FROM scratch
+
+COPY --from=build /usr/bin/ls /usr/bin/ls
```
Comment (razvand) [inline examples/core-utils/Dockerfile:5]:
```suggestion
# Binaries
COPY --from=build /usr/bin/ls /usr/bin/ls 
```

File: examples/core-utils/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM debian:bookworm AS build 
+
+FROM scratch
+
+COPY --from=build /usr/bin/ls /usr/bin/ls 
+COPY --from=build /usr/bin/cat /usr/bin/cat  
+COPY --from=build /usr/bin/pwd /usr/bin/pwd
+COPY --from=build /usr/bin/uniq /usr/bin/uniq
+# Dependencies
```
Comment (razvand) [inline examples/core-utils/Dockerfile:9]:
```suggestion

# Library dependencies
```

File: examples/core-utils/Kraftfile.cat [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      CONFIG_LWIP_SOCKET: 'y'
+      CONFIG_LWIP_AUTOIFACE: 'y'
+      CONFIG_LWIP_IPV4: 'y'
+      CONFIG_LWIP_DHCP: 'y'
+      CONFIG_LWIP_DNS: 'y'
+      CONFIG_LWIP_NUM_TCPCON: 64
+      CONFIG_LWIP_NUM_TCPLISTENERS: 64
+      CONFIG_LWIP_ICMP: 'y'
+  libelf:
+    source: https://github.com/unikraft/lib-libelf.git
```
Comment (razvand) [inline examples/core-utils/Kraftfile.cat:139]:
```suggestion
runtime: base:latest
```

File: examples/core-utils/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Linux Core Utils
+
+This directory contains a minimal Debian instance featuring ore utils such as ls, cat, pwd, etc., running on Unikraft.
+
+## Set Up
+
+Use `kraft` to run the image and start a Unikraft instance:
+
+```bash
```
Comment (razvand) [inline examples/core-utils/README.md:9]:
```suggestion
```console
```

File: examples/core-utils/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# Linux Core Utils
```
Comment (razvand) [inline examples/core-utils/README.md:1]:
```suggestion
# Linux Coreutils
```

File: examples/core-utils/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+This directory contains a minimal Debian instance featuring ore utils such as ls, cat, pwd, etc., running on Unikraft.
+
+## Set Up
+
+Use `kraft` to run the image and start a Unikraft instance:
+
+```bash
+kraft run --rm --plat qemu --arch x86_64 -K Kraftfile.{`cmd`} .
+```
+Where `cmd` is the command you want to execute.
```
Comment (razvand) [inline examples/core-utils/README.md:12]:
```suggestion

Where `cmd` is the command you want to execute.
```

**Review by razvand — CHANGES_REQUESTED** [review]
Apply updates to all `Dockerfile`s as in the feedback (i.e. use `runtime:main`).

Squash the two commits in a single one.

Rename directory to `coreutils/` instead of `core-utils`.

See my other comments.

File: examples/core-utils/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# Linux Coreutils
+
+This directory contains a minimal Debian instance featuring ore utils such as ls, cat, pwd, etc., running on Unikraft.
```
Comment (razvand) [inline examples/core-utils/README.md:3]:
```suggestion
This directory contains a minimal Debian instance featuring core utilities such as `ls`, `cat`, `pwd`, etc., running on Unikraft.
```

File: examples/core-utils/Kraftfile.ls [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+      CONFIG_LWIP_AUTOIFACE: 'y'
+      CONFIG_LWIP_IPV4: 'y'
+      CONFIG_LWIP_DHCP: 'y'
+      CONFIG_LWIP_DNS: 'y'
+      CONFIG_LWIP_NUM_TCPCON: 64
+      CONFIG_LWIP_NUM_TCPLISTENERS: 64
+      CONFIG_LWIP_ICMP: 'y'
+  libelf:
+    source: https://github.com/unikraft/lib-libelf.git
+    version: staging
```
Comment (razvand) [inline examples/core-utils/Kraftfile.ls:140]:
```suggestion
runtime: base:latest
```

File: examples/core-utils/Kraftfile.pwd [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+      CONFIG_LWIP_NUM_TCPCON: 64
+      CONFIG_LWIP_NUM_TCPLISTENERS: 64
+      CONFIG_LWIP_ICMP: 'y'
+  libelf:
+    source: https://github.com/unikraft/lib-libelf.git
+    version: staging
+
+targets:
+- fc/x86_64
+- qemu/x86_64
```
Comment (razvand) [inline examples/core-utils/Kraftfile.pwd:144]:
```suggestion
runtime: base:latest
```

File: examples/core-utils/Kraftfile.uniq [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+      CONFIG_LWIP_NUM_TCPCON: 64
+      CONFIG_LWIP_NUM_TCPLISTENERS: 64
+      CONFIG_LWIP_ICMP: 'y'
+  libelf:
+    source: https://github.com/unikraft/lib-libelf.git
+    version: staging
+
+targets:
+- fc/x86_64
+- qemu/x86_64
```
Comment (razvand) [inline examples/core-utils/Kraftfile.uniq:144]:
```suggestion
runtime: base:latest
```

File: examples/core-utils/Kraftfile.cat [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+rootfs: ./Dockerfile
+
+cmd: ["/usr/bin/cat"]
+
+runtime: base:latest
+    version: staging
+
+targets:
+- fc/x86_64
+- qemu/x86_64
```
Comment (razvand) [inline examples/core-utils/Kraftfile.cat:14]:
```suggestion
```

File: examples/core-utils/.gitignore [hunk 12]
```diff
@@ -0,0 +1 @@
+*.txt
```
Comment (razvand) [inline examples/core-utils/.gitignore:1]: What is the point of this?

**Review by razvand — CHANGES_REQUESTED** [review]
Apply updates to all `Dockerfile`s as in the feedback (i.e. use `runtime: base:latest`).

Squash all commits into a single commit.

Rename directory to `coreutils/` instead of `core-utils/`.

File: examples/coreutils/Kraftfile.ls [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+
+rootfs: ./Dockerfile
+
+cmd: ["/usr/bin/ls"]
+
+runtime: base:latest
+
+targets:
+- fc/x86_64
+- qemu/x86_64
```
Comment (razvand) [inline examples/coreutils/Kraftfile.ls:13]:
```suggestion
```

File: examples/coreutils/Kraftfile.cat [hunk 14]
```diff
@@ -0,0 +1,9 @@
+spec: v0.6
+
+name: cat 
+
+rootfs: ./Dockerfile
+
+cmd: ["/usr/bin/cat"]
```
Comment (razvand) [inline examples/coreutils/Kraftfile.cat:7]: Update the command to list the contents of a file. Add some files to the local directory in the repository than copy them to the `Dockerfile`:

```Dockerfile
COPY myfs/ .
```

File: examples/coreutils/Kraftfile.uniq [hunk 15]
```diff
@@ -0,0 +1,9 @@
+spec: v0.6
+
+name: uniq 
+
+rootfs: ./Dockerfile
+
+cmd: ["/usr/bin/uniq"]
```
Comment (razvand) [inline examples/coreutils/Kraftfile.uniq:7]: Also pass an argument to it from the local filesystem, e.g. `cmd: ["/usr/bin/uniq", "/file.txt"]`

File: examples/coreutils/README.md [hunk 16]
```diff
@@ -0,0 +1,18 @@
+# Linux Coreutils
+
+This directory contains a minimal Debian instance featuring core utilities such as `ls`, `cat`, `pwd`, etc., running on Unikraft.
+
+## Set Up
+
+Use `kraft` to run the image and start a Unikraft instance:
+
+```console
+kraft run --rm --plat qemu --arch x86_64 -K Kraftfile.{`cmd`} .
```
Comment (razvand) [inline examples/coreutils/README.md:10]:
```suggestion
kraft run --rm --plat qemu --arch x86_64 -K Kraftfile.<cmd> .
```

File: examples/coreutils/README.md [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+```
+
+Where `cmd` is the command you want to execute.
```
Comment (razvand) [inline examples/coreutils/README.md:13]:
```suggestion
Where `<cmd>` is the command you want to execute.
```

**Review by razvand — CHANGES_REQUESTED** [review]
See my comments.

Update your commit such that the signed-off name is similar to the one in the signature, i.e.:

```
Signed-off-by: George-Theodor Paraschiva <paraschiva.george.theodor@gmail.com>
```

Update the commit message to detail the tools / utils that you added in the catalog.

**Review by razvand — CHANGES_REQUESTED** [review]
Just some small comments and then we can merge.

**Review by razvand — APPROVED** [review]
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request merged**